include/aws/http/private/h2_connection.h

```diff
@@ -125,6 +125,13 @@ struct aws_h2_connection {
         uint64_t outgoing_timestamp_ns;
         /* Timestamp when connection has data to receive, which is when there is an active stream */
         uint64_t incoming_timestamp_ns;
+
+        /**
+         * Contains struct aws_http_stream *
+         * The streams that encoded the last frame between it's encoded to the `aws_io_message` to the
+         *`aws_io_message.on_completion` invoked.
+         **/
+        struct aws_array_list finish_encoding_streams_list;
```

Comment:
IMHO, the extra precision doesn't seem worth all the complexity and potential bugs

Reply:
Like, maybe we just wrap the input_stream, first call to read() sets start time, and when get_status() says it's done, that's the end time

Reply:
knowing the write completion callbacks is ... nice. But the complexity and weirdness they introduce make it so scary

Reply:
Get back to the simpler solution to track the time that stream thinks it starts/ends.